src/Grid/Service/ConfigurationService.php

```diff
@@ -0,0 +1,94 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Bundle\StudioBackendBundle\Grid\Service;
+
+use Pimcore\Bundle\StudioBackendBundle\Grid\Event\GridColumnConfigurationEvent;
+use Pimcore\Bundle\StudioBackendBundle\Grid\Schema\ColumnConfiguration;
+use Pimcore\Bundle\StudioBackendBundle\Util\Constants\ElementTypes;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use function count;
+use function in_array;
+
+/**
+ * @internal
+ */
+final readonly class ConfigurationService implements ConfigurationServiceInterface
+{
+    public function __construct(
+        private GridServiceInterface $gridService,
+        private EventDispatcherInterface $eventDispatcher,
+        private array $predefinedColumns
+    ) {
+    }
+
+    /**
+     * @return ColumnConfiguration[]
+     */
+    public function getAvailableAssetGridConfiguration(): array
+    {
+        $columns = [];
+        foreach ($this->gridService->getColumnCollectors() as $collector) {
+            // Only collect supported asset collectors
+            if (!in_array(ElementTypes::TYPE_ASSET, $collector->supportedElementTypes(), true)) {
+                continue;
+            }
+
+            $columns = array_merge(
```

Comment:
this can be replaced by the spread operator like here https://github.com/pimcore/studio-backend-bundle/pull/308/files#diff-614133628b0e63d8f255992a27087738b4073af6b3e1cccd5ce6d886bbace99dR159

Array merge is very greedy, even PHPstorm complains :D

Reply:
I guess with the merge of the export, you have to check this anyways again. Sorry :)

Reply:
Yeah good point... I moved it from the GridService... but i will change it.

Reply:
Fixed https://github.com/pimcore/studio-backend-bundle/pull/311/commits/bd0ded320676342dc5acda928cdbb827c3a489ac